### PR TITLE
[Data History Explorer]: Telemetry and Logging

### PR DESCRIPTION
**Pull request by iremgokce**
### Summary of changes 🔍 
- Added telemetry for Data History Explorer (for both system and user actions) and logging of actions. Note that I added a separate section for `DataHistoryExplorer` in the `TelemetryConstants` file since it is a independent entity as an orthogonal tool to the 3D scene app (different than Adapters, Builder and Viewer app regions). If you have concerns about it we can discuss more.
- **To-do:** I am going to emit `OpenModal` user action telemetry when I bind the modal to property inspector soon.

image
image

Here is some examples:
image

### Testing 🧪

1. Enable telemetry logging feature in your local storage using `localStorage.setItem('cardboard.debug.telemetryLogging',true)`;
2. Enable "Verbose" to see the telemetry logging in the inspector on the page
3. Go to Components/DataHistoryExplorer/Modal ADT story and see the logs and telemetry

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [x] Added relevant labels
- [x] Requested developer reviews
- [ ] Request UI review from design / PM
- [x] Verify all code checks are passing

### CODE REVIEW

File: src/Models/Constants/TelemetryConstants.ts [hunk 1]
```diff
@@ -177,7 +179,81 @@ export const TelemetryEvents = {
...
     },
     /** viewer section of the app */
-    Viewer: {}
+    Viewer: {},
+    /** tools provided by the library */
+    Tools: {
+        /** data history explorer component */
+        DataHistoryExplorer: {
+            SystemAction: {
+                ParseModal: {
```
Comment (pasanchMSFT) [inline src/Models/Constants/TelemetryConstants.ts:188]: Suggestion for a name change: `ParseDataHistory`, Modal is not very representative of the data.

File: src/Models/SharedUtils/DataHistoryUtils.ts [hunk 2]
```diff
@@ -176,3 +184,46 @@ export const getDefaultSeriesLabel = (
...
+            customProperties: {
+                [event.properties.countSeries]: timeSeriesTwinList.length
+            },
+            triggerType: TelemetryTrigger.SystemAction
+        })
+    );
+};
+
+/** send the user actions telemetry  */
+export const sendDataHistoryExplorerUserTelemetry = (
```
Comment (pasanchMSFT) [inline src/Models/SharedUtils/DataHistoryUtils.ts:209]: I like this pattern for sending events that have the same keys in custom properties.

**Review by pasanchMSFT — APPROVED** [review]
LGTM

File: src/Models/Constants/TelemetryConstants.ts (hunk 1 above)
Comment (iremgokce) [reply src/Models/Constants/TelemetryConstants.ts:188]: make sense, changing it

**Review by montgomp — APPROVED** [review]
👍🏻LGTM

**Pull request merged**